### ISSUE TITLE
Bump ESLint major version in Peer Dependencies (eslint 6.0)

### ISSUE THREAD

**Issue by SalvatorePreviti** [typescript-eslint/typescript-eslint#644]
**What code were you trying to parse?**

`npm install`

**What did you expect to happen?**

no warnings

**What actually happened?**

```
npm WARN @typescript-eslint/eslint-plugin@1.11.0 requires a peer of eslint@^5.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN @typescript-eslint/parser@1.11.0 requires a peer of eslint@^5.0.0 but none is installed. You must install peer dependencies yourself.
```

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | 1.11.0 |
| `TypeScript`                | latest |
| `ESLint`                    | `6.0.0` |
| `node`                      | `12.4.0` |
| `npm`                       | `6.9.0` |

**Comment by bradzacher** [comment]
We'll bump it as soon as we start testing against it.
We need to review the release notes and action things appropriately.

For posterity's sake: https://eslint.org/docs/6.0.0/user-guide/migrating-to-6.0.0

**Comment by mrmckeb** [comment]
I did a quick check and it seems that the parser relies on `eslint/lib/util/ast-utils` which seems to have been removed from ESLint.

**Comment by bradzacher** [comment]
@mrmckeb - no an in-development, unreleased rule depends on that file, but it's not requried anywhere and not exposed to users, so it doesn't matter.

We already fixed the file dependency issues before the 6.0 release because some users were using the beta releases and reported issues.

This plugin will probably "just work" with eslint v6.
But until we've reviewed the changelog accordingly and tested it, we won't be bumping the version range.
